algorithm_services/algorithms/__init__.py

```diff
@@ -3,7 +3,8 @@
 
 class Algorithm():
 
-    def __init__(self, args):
+    def __init__(self, name, args):
```

Comment:
Use class attr instead of argument:

class Algorithm:
    name = “Undefined”
    ....

 At Algorithm classes you should overwrite:

class FizzBuzz(Algorithm):
    name = “FizzBuzz”

Or you can set a property to retrieve class directly:

class Algorithm:

    @property
    def name(self):
        return self.__name__